filesystem/iso9660.ksy

```diff
@@ -5,269 +5,684 @@ meta:
+        type: datetime_short
+        if: _parent.effective
+  rrip_tf_long:
+    doc-ref: rrip 4.6.1
+    doc: |
+      The data is available in this specific order.
+      Multiple dates can be set.
+    seq:
+      - id: creation
+        type: datetime_long
+        if: _parent.creation
+      - id: modify
+        type: datetime_long
+        if: _parent.modify
+      - id: access
+        type: datetime_long
+        if: _parent.access
+      - id: attributes
+        type: datetime_long
+        if: _parent.attributes
+      - id: backup
+        type: datetime_long
+        if: _parent.backup
+      - id: expiration
+        type: datetime_long
+        if: _parent.expiration
+      - id: effective
+        type: datetime_long
+        if: _parent.effective
+  rrip_tf:
+    doc-ref: rrip 4.6.1
+    seq:
+      - id: length
         type: u1
-      - id: minute
+      - id: version
+        contents: [ 0x1 ]
+      - id: long_form
+        type: b1
+      - id: effective
+        type: b1
+      - id: expiration
+        type: b1
+      - id: backup
+        type: b1
+      - id: attributes
+        type: b1
+      - id: access
+        type: b1
+      - id: modify
+        type: b1
+      - id: creation
+        type: b1
+      - id: datetime_short
+        type: rrip_tf_short
+        if: not long_form
+      - id: datetime_long
+        type: rrip_tf_long
+        if: long_form
+  susp_unknown: # default for now
+    seq:
+      - id: length
         type: u1
-      - id: sec
+      - id: version
         type: u1
-      - id: timezone
+      - id: data
+        size: length - 4
+  su_header:
+    seq:
+      - id: signature
+        type: u2be
+        enum: su_signature
+      - id: su_specific
+        type:
+          switch-on: signature
+          cases:
+            'su_signature::rras_amiga_specific': susp_unknown # AS
+            'su_signature::susp_continuation_area': susp_ce # CE
+            'su_signature::rrip_child_link': susp_unknown # CL
+            'su_signature::susp_extensions_reference': susp_er # ER
+            'su_signature::susp_extension_selector': susp_es # ES
+            'su_signature::rrip_alternate_name': rrip_nm # NM
+            'su_signature::susp_padding_field': susp_pd # PD
+            'su_signature::rrip_parent_link': susp_unknown # PL
+            'su_signature::rrip_posix_device_number': susp_unknown # PN
+            'su_signature::rrip_posix_file_attributes': rrip_px # PX
+            'su_signature::rrip_relocated_directory': susp_unknown # RE
+            'su_signature::rrip_extensions_in_use_indicator': susp_unknown # RR
+            'su_signature::rrip_sparse_file': susp_unknown # SF
+            'su_signature::rrip_symbolic_link': susp_unknown # SL
+            'su_signature::susp_indicator': susp_sp # SP
+            'su_signature::susp_terminator': susp_st # ST
+            'su_signature::rrip_time_file': rrip_tf # TF
+            'su_signature::rrzf_zisofs': susp_unknown # ZF
```

Comment:
Aren't they subtypes? Should we put them into own ```types``` and drop the prefix?

Reply:
Making sure I understand your comment:

I put the prefix in front of it as a reference to the specification.
As I also put that in the doc-ref, I should drop that.

Reply:
Not quite.

I mean that it often happens that some types are used only in some type. In this case to improve readability (most of editors allow to fold all the blocks and then only keep the relevant blocks unfolded) we can move it into `types` in the type using these types. Often these types have a prefix in name of the name type they are used in. When inlined their "address" would be ```<parent_type_name>::<parent_type_name>_<subtype_name>```. It's ugly and redundant. In these cases I remove parent type name making the path look like ```<parent_type_name>::<subtype_name>```.

Reply:
I am sorry, but I really do not understand this comment...
Are you talking about: 
- a nested enum
- splitting the enum, and reference multiple?
- moving the enum into su_header
Or just something else?

Reply:
I made the enums local

Reply:
I meant not only enums, but the types too.

Reply:
Okay

Reply:
Done